guides/micronaut-oracle-autonomous-db/micronaut-oracle-autonomous-db.adoc

```diff
@@ -0,0 +1,235 @@
+
+image::oracleautodb/autonomous.database.4.png[]
+
+Select "License Included" and click "Create Autonomous Database" to create your instance:
+
+image::oracleautodb/autonomous.database.5.png[]
+
+On the "Autonomous Database Details" page click the "Copy" link in the `OCID` row; this is the unique identifier for your database instance and you'll need it later in your application's `application.yml`.
+
+image::oracleautodb/autonomous.database.6.png[]
+
+Next, create a schema user. On the "Autonomous Database Details" page select the "Tools" tab and click "Open Database Actions":
+
+image::oracleautodb/autonomous.database.7.png[]
+
+Login with username "ADMIN" and the admin password you defined earlier:
+
+image::oracleautodb/autonomous.database.8.png[]
+
+Under "Development" click the "SQL" button to open the SQL console:
+
+image::oracleautodb/autonomous.database.9.png[]
+
+Copy and paste the following SQL which will create a schema user into the worksheet:
+
+```
+CREATE USER micronautdemo IDENTIFIED BY "XXXXXXXXX";
+GRANT CONNECT, RESOURCE TO micronautdemo;
+GRANT UNLIMITED TABLESPACE TO micronautdemo;
+```
+
+Create a schema user password (must be at least 12 characters and contain a number and an uppercase letter) and replace the text "XXXXXXXXX" with that password.
+
+Click the image:oracleautodb/runscript.png[] ("Run Script") button to execute the SQL:
+
+image::oracleautodb/autonomous.database.10.png[]
+
+== Writing the App
+
+include::{commondir}/common-cli-or-launch.adoc[]
+
+[source,bash]
+----
+mn create-app example.micronaut.micronautguide --build=@build@ --lang=@lang@ --jdk=11 --features=oracle,data-jdbc,oracle-cloud-sdk,flyway
+----
+
+include::{commondir}/common-build-lang-arguments.adoc[]
+
+include::{commondir}/common-default-package.adoc[]
+
+include::{commondir}/common-annotationprocessors.adoc[]
+
+=== Dependencies
+
+Oracle Cloud Autonomous Database connection information and credentials are stored in the https://docs.oracle.com/en-us/iaas/Content/Database/Tasks/adbconnecting.htm[Oracle Wallet]. In addition to the dependencies added by the features specified when creating the app, we need a few more for Autonomous Database connectivity and automatic wallet configuration:
+
+dependency:micronaut-oraclecloud-atp[groupId=io.micronaut.oraclecloud]
+
+:exclude-for-build:maven
+[source,groovy]
+----
+implementation platform("com.oracle.database.jdbc:ojdbc-bom:21.1.0.0")
+----
+:exclude-for-build:
+
+:exclude-for-build:gradle
+[source,xml]
+----
+<dependencyManagement>
+    <dependencies>
+        <dependency>
+            <groupId>com.oracle.database.jdbc</groupId>
+            <artifactId>ojdbc-bom</artifactId>
+            <version>21.1.0.0</version>
+            <type>pom</type>
+            <scope>import</scope>
+        </dependency>
+    </dependencies>
+</dependencyManagement>
+----
+:exclude-for-build:
```

Comment:
Don't define external dependencies manually in the guide. Create a `Feature` like these: https://github.com/micronaut-projects/micronaut-guides/tree/master/buildSrc/src/main/java/io/micronaut/guides/feature and add the dependency in [pom.xml](https://github.com/micronaut-projects/micronaut-guides/blob/master/buildSrc/src/main/resources/pom.xml) so the versions get updated automatically.

Then you can include that new feature in the guide `metadata.json` so the generated project contains the appropriate dependencies.

Also keep in mind that you instead of hardcoding the version in the guide you can do it as explained in this PR: https://github.com/micronaut-projects/micronaut-guides/pull/225

Reply:
That doesn't seem to support bom dependencies, should I add that to starter or am I missing the syntax for `implementation(platform("..."))`?

Reply:
You need the feature in any case because that's the only way to add dependencies to the generated projects. Without the dependencies declared in `metadata.json` the generated projects won't have the necessary dependencies and the apps won't compile.